glotaran/project/dataclasses.py

```diff
@@ -0,0 +1,125 @@
+"""Contains helper methods for dataclasses."""
+from __future__ import annotations
+
+import dataclasses
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Callable
+    from typing import TypeVar
+
+    DefaultType = TypeVar("DefaultType")
+
+
+def exclude_from_dict_field(
```

Comment:
See other comments. Also consider adding some documentation as to _why_ one might want to do this.

Reply:
We should leave that to follow up PR. This isn't the final incarnatian just a basic idea to make it work.